evalml/pipelines/pipeline_base.py

```diff
@@ -190,17 +190,24 @@ def compute_estimator_features(self, X):
         """
         X_t = X
         for component in self.component_graph[:-1]:
-            X_t = component.transform(X_t)
+            X_t = component.transform(X_t, y)
         return X_t
 
-    def _fit(self, X, y):
+    def _compute_features_during_fit(self, X, y):
```

Comment:
Need to make this small refactor because we need to make sure the Nans are dropped before the estimator is fit. I think it makes the most sense to do that in the time series pipeline as opposed to the base class.